### PR TITLE
Gary/feature/authentication

### PR DESCRIPTION
**Pull request by garysmith1933**
# Description
This PR includes migration of the mongdb database over to supabase and implementation of supabase authentication. As a result, a user can now signup, login, and logout of the application. The user can also see their data currently on the /private page when logged in.

## Related to the following Jira/Trello issue 
- https://trello.com/c/F0fho90n/37-user-login
- https://trello.com/c/cM8UbjdC/40-create-a-database-to-store-resident-contact-information

## Main changes explained:
- Removed DBConnect.js, Mongodb schemas as it is no longer needed.
- Refactored Seed.js to handle implementing dummy data to supabase database.
- Added supabase authentication, middleware to handle requests and keep track of cookies.
- Refactored login page to implement login functionality.
- Created signup form to implement signup functionality.
- Adjusted logout button in header component to implement logout functionality.
- Created private/page.js to see data of logged in user.

## How to test:
1. check into current branch
2. do `npm install` and `npm run dev` to run this PR locally
3. Verify that you added the new environment variables to the .env and .env.local files. They can be found in the environment variables discussion in the discord.
4. When starting the application, click the signup button, this should lead you to the sign up form to enter your info to sign up for the application.
5. The page should lead you back to the '/' page, or the '/error' page if something went wrong. To check that you are logged in head over to localhost:3000/private. If you are logged in successfully you should see your email, name, address, and phone number.
6. Test logging out of the application by clicking the logout button, this should redirect you to the '/' page or the '/error page if something went wrong. Verify that you are logged out by trying to visit the '/private' page again, this time you should be redirected over to the login page.
7. Once you are on the login page try to sign back in to the application with your email and password. If successful it should redirect you to the home page and the you should be able to acess the private page once again to see your data.

## Note:
There are little improvements that can be made here such as having a modal up to confirm if a user wants to sign out, or having errors display if an input is empty or invalid in the sign up and sign in forms but will be addressed at a later time. 

Seed.js will also need to be slighly adjusted to handle signing up users. Will handle that tomorrow.

I will also add comments to explain what does what.

### CODE REVIEW

File: src/app/private/page.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { redirect } from 'next/navigation'
```
Comment (RossaMania) [inline src/app/private/page.js:1]: PRAISE: Nice! I like that this is a "read view" of the user data that was put in on signup.

QUESTION: Is this just supposed to be temporary skeleton we can use for the "/user" route to prove that the sign up feature is functioning?

Or is this meant as a foundation to start to replace the "/user" route?

File: src/app/layout.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import localFont from "next/font/local";
 import "./globals.css";
+import 'bootstrap/dist/css/bootstrap.min.css';
```
Comment (RossaMania) [inline src/app/layout.js:3]: PRAISE: Great job figuring out what file we're supposed to import bootstrap into!

File: src/app/private/page.js (hunk 1 above)
Comment (garysmith1933) [reply src/app/private/page.js:1]: Its just to prove that sign up and login are working. It was shown in the documentation so I just built a little bit on top of it by adding some additional info.

Comment (RossaMania) [reply src/app/private/page.js:1]: Great! That's exactly what I was thinking!

**Comment by RossaMania** [comment]
image
This pull request looks good to me overall!

I can wait until this stuff gets done before approving and merging!

**Review by RossaMania — APPROVED** [review]
This is a great start to getting authentication working on the app! Great job swapping out the database!

**Pull request merged**